src/Tempest/Filesystem/src/ErrorContext.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Tempest\Filesystem;
+
+final class ErrorContext
```

Comment:
@innocenzi - Can I get a quick sanity check here? Is this helpful or an unnecessary abstraction?

Reply:
If the functions were `try`/`catch`able, I'd prefer that, but since you need to call `error_get_last()`, you might as well have a nice experience with that

However, I'm wondering if `error_clear_last()` is really needed? Since you check for the result of warning-emitting functions, `error_get_last()` will always be the relevant error.

So maybe simplify `ErrorContext` to have a sane way of retrieving the last error message, without needing for the reset part?

Reply:
Yeah this is good 👍